index.js

```diff
@@ -1,119 +1,157 @@
+validateInput(passedCountry,passedProductType);
 
-      const name = store.storeName;
-      let productStatus = [];
+const countryCode = COUNTRIES[passedCountry];
+if (countryCode === "/au") {
+  skuList = passedProductType === PRODUCT_TYPES.IPHONE ? SKUS_AUSTRALIA_IPHONE : SKUS_AUSTRALIA;
+  favorites = favoritesAustralia;
+}
 
-      for (const [key, value] of Object.entries(skuList)) {
-        const product = store.partsAvailability[key];
+// run the search
+queryStore(countryCode,passedStore,skuList,favorites);
 
-        hasStoreSearchError = product.storeSearchEnabled !== true;
+// #region Helper Functions
 
-        if (key === control && hasStoreSearchError !== true) {
-          hasStoreSearchError = product.pickupDisplay !== "available";
-        } else {
-          productStatus.push(`${value}: ${product.pickupDisplay}`);
+function validateInput(propCountry,propProductType) {
+  if (!Object.getOwnPropertyNames(COUNTRIES).includes(propCountry)) {
+    console.error("Unrecognised country code");
+    console.error(`Recognised codes: ${Object.getOwnPropertyNames(COUNTRIES).join(',')}`);
+    process.exit(1);
+  }
+  if (!Object.getOwnPropertyNames(PRODUCT_TYPES).includes(propProductType)) {
+    console.error("Unrecognised product type");
+    console.error(`Recognised product types: ${Object.getOwnPropertyNames(PRODUCT_TYPES).join(',')}`);
+    process.exit(2);
+  }
+}
 
-          if (product.pickupDisplay !== "unavailable") {
-            console.log(`${value} in stock at ${store.storeName}`);
-            let count = skuCounter[key] ?? 0;
-            count += 1;
-            skuCounter[key] = count;
+function queryStore(country, storeNumber, skuList, favorites) {
+  const query =
+    Object.keys(skuList)
+      .map((k, i) => `parts.${i}=${encodeURIComponent(k)}`)
+      .join("&") + `&searchNearby=true&store=${storeNumber}`;
+
+  let options = {
+    method: "GET",
+    url: `https://www.apple.com${country}/shop/fulfillment-messages?` + query,
+  };
+
+  request(options, (error, response) => {
+    if (error) throw new Error(error);
+
+    const body = JSON.parse(response.body);
+    const storesArray = body.body.content.pickupMessage.stores;
+    let skuCounter = {};
+    let hasStoreSearchError = false;
+
+    console.log('Inventory');
+    console.log('---------');
+    const statusArray = storesArray
+      .flatMap((store) => {
+        const name = store.storeName;
+        let productStatus = [];
+
+        for (const [key, value] of Object.entries(skuList)) {
+          const product = store.partsAvailability[key];
+
+          hasStoreSearchError = product.storeSearchEnabled !== true;
+
+          if (key === control && hasStoreSearchError !== true) {
+            hasStoreSearchError = product.pickupDisplay !== "available";
+          } else {
+            productStatus.push(`${value}: ${product.pickupDisplay}`);
+
+            if (product.pickupDisplay !== "unavailable") {
```

Comment:
not sure if this would be obvious in the conflict, but I changed this to check for `=== "available"` instead - only checking for `unavailable` led to some inconsistent results in countries where some products return `ineligible` here, so checking for the positive case instead leads to more consistent output.

Reply:
Thanks for the heads up :)

Regarding pulling the SKUs, I was able to do that by going to the iphone 13 / 13pro page and running this JS:

```javascript
temp = []; window.PRODUCT_SELECTION_BOOTSTRAP.productSelectionData.products.map((a)=> temp[`${a.partNumber}`] = `${a.productLocatorFamily} ${a.dimensionCapacity} ${a.dimensionColor}`); 
temp
```

One issue I need to check is whether all iphone SKUs can have a country code swapped. I'm pretty sure I saw at least one that didn't fit that pattern. I'll check today.